src/SFA.DAS.EAS.Portal/Database/Models/Account.cs

```diff
@@ -0,0 +1,126 @@
+
+        private Account(long accountId, DateTime created, string messageId)
+            : base(1)
+        {
+            Id = Guid.NewGuid();
+            AccountId = accountId;
+            // the created date originates from event publishers, so we make sure we store it as UTC
+            Created = DateTime.SpecifyKind(created, DateTimeKind.Utc);
+            
+            AddOutboxMessage(messageId, created);
+
+            Organisations = new List<Organisation>();
+        }
+
+        // ctor for when creating account doc from reserved funding
+        // do we accept event, seems a coupling too far!
+        public Account(long accountId, long accountLegalEntityId, string legalEntityName, Guid reservationId,
+            string courseId, string courseName, DateTime startDate, DateTime endDate, DateTime created, string messageId)
+            : this(accountId, created, messageId)
+        {
+            Organisations = new List<Organisation>();
+            AddReserveFunding(accountLegalEntityId, legalEntityName, reservationId, courseId, courseName, startDate, endDate);
+        }
+
+        [JsonConstructor]
+        public Account()
+        {
+            Organisations = new List<Organisation>();
+        }
+
+        public void AddReserveFunding(long accountLegalEntityId, string legalEntityName,  Guid reservationId,
+            string courseId, string courseName, DateTime startDate, DateTime endDate, DateTime updated, string messageId)
+        {
+            ProcessMessage(messageId, updated, () =>
+            {
+                var existingAccountLegalEntity = _accountLegalEntities.FirstOrDefault(ale => ale.AccountLegalEntityId == accountLegalEntityId);
+                if (existingAccountLegalEntity == null)
+                    AddReserveFunding(accountLegalEntityId, legalEntityName, reservationId, courseId, courseName, startDate, endDate);
+                else
+                {
+                    existingAccountLegalEntity.AddReserveFunding(reservationId, courseId, courseName, startDate, endDate);
+                }
+                
+                Updated = updated;
+                Deleted = null;
+            });
+        }
+
+        private void AddReserveFunding(long accountLegalEntityId, string legalEntityName,
+            Guid reservationId, string courseId, string courseName, DateTime startDate, DateTime endDate)
+        {
+            _accountLegalEntities.Add(new AccountLegalEntity(accountLegalEntityId, legalEntityName, reservationId, courseId, courseName, startDate, endDate));
+        }
+
+        internal void AddOutboxMessage(string messageId, DateTime created)
+        {
+            if (messageId == null)
+                throw new ArgumentNullException(nameof(messageId));
+            
+            _outboxData.Add(new OutboxMessage(messageId, created));
+        }
+
+        // todo: it would be better to have a webjob to periodically clean-up, but for now..
```

Comment:
Remove the todo?  A web job would be over engineering perhaps?  Having the code self clean is simpler and more reliable.

Reply:
The issue with the current scheme, is that the `Account` document is only cleaned up when the document is touched. If the account is never touched, the outdated messages will remain forever.

Having a WebJob perform the clean up, would mean that old messages could be deleted, irrespective of the account not being touched.

Reply:
Does it matter if they are not deleted?  They are not increasing in size.

Reply:
That's why the current scheme is good enough (especially for MVS), but it would be tidier to not store data that is of no use to man nor beast.

Reply:
File has been deleted. Note added to dev notes file.